### PR TITLE
Add LocalTime & Maybe SDK Functions

### PR DESCRIPTION
**Pull request by sahilravinder**
- MaybeSDK.map2
- MaybeSDK.map3
- MaybeSDK.map4
- MaybeSDK.hasValue
- LocalTimeSDK.diffInHours
- LocalTimeSDK.diffInMinutes

### CODE REVIEW

File: morphir/src/org/finos/morphir/runtime/sdk/MaybeSDK.scala [hunk 1]
```diff
@@ -74,4 +76,56 @@ object MaybeSDK {
         resultToMaybe(out)
       }
   }
+
+  val map2 = DynamicNativeFunction3("map2") {
+    (ctx: NativeContext) => (f: RT.Function, maybeRaw1: RT.ConstructorResult, maybeRaw2: RT.ConstructorResult) =>
+      {
+        val out = for {
+          elem1 <- eitherToOption(maybeRaw1)
```
[the remark this answers is not in the document]
Comment (DamianReeves) [reply morphir/src/org/finos/morphir/runtime/sdk/MaybeSDK.scala:84]: There's part of this implementation that feels strange, but I'm a bit swamped to review this right now... (@michelchan , @edwardpeters , or @Paul-Folbrecht ) can any of you take a look at this?

[the remark this answers is not in the document]
Comment (michelchan) [reply morphir/src/org/finos/morphir/runtime/sdk/MaybeSDK.scala:84]: Agreed. The comments say `ConstructorResult` is used for Maybes and Results, and neither are an `Either`. So this should at least be renamed.

File: morphir/runtime/test/jvm/src/org/finos/morphir/runtime/EvaluatorMDMTests.scala [hunk 2]
```diff
@@ -1199,6 +1199,20 @@ object EvaluatorMDMTests extends MorphirBaseSpec {
         testEvalMultiple("diffInYears")("localDateTests", "diffInYearsTest", List(localDate, localDate.plusYears(2)))(
           Data.Int(2)
         ),
+        testEvalMultiple("diffInYearsAdjacent")(
```
Comment (michelchan) [inline morphir/runtime/test/jvm/src/org/finos/morphir/runtime/EvaluatorMDMTests.scala:1202]: For consistency, can you update the elm test examples with these test cases? Also add them for the LocalTime ones you added.

File: morphir/runtime/test/jvm/src/org/finos/morphir/runtime/EvaluatorMDMTests.scala [hunk 3]
```diff
@@ -1199,6 +1199,20 @@ object EvaluatorMDMTests extends MorphirBaseSpec {
...
+          "localDateTests",
+          "diffInYearsTest",
+          List(java.time.LocalDate.of(1900, 12, 31), java.time.LocalDate.of(1901, 1, 1))
+        )(
+          Data.Int(1)
+        ),
+        testEvalMultiple("diffInYearsFullYear")(
```
Comment (michelchan) [inline morphir/runtime/test/jvm/src/org/finos/morphir/runtime/EvaluatorMDMTests.scala:1209]: Would rename this to `diffInYearsSameYear`

Comment (sahilravinder) [reply morphir/runtime/test/jvm/src/org/finos/morphir/runtime/EvaluatorMDMTests.scala]: fixed

File: morphir/runtime/test/jvm/src/org/finos/morphir/runtime/EvaluatorMDMTests.scala (hunk 2 above)
Comment (sahilravinder) [reply morphir/runtime/test/jvm/src/org/finos/morphir/runtime/EvaluatorMDMTests.scala:1202]: fixed

File: morphir/src/org/finos/morphir/runtime/sdk/MaybeSDK.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (DamianReeves) [reply morphir/src/org/finos/morphir/runtime/sdk/MaybeSDK.scala:84]: Added #575 to track this

**Review by DamianReeves — APPROVED** [review]
I'm good with this, but would like to see #575 addressed as mentioned in the comments

File: morphir/src/org/finos/morphir/runtime/sdk/MaybeSDK.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (edwardpeters) [reply morphir/src/org/finos/morphir/runtime/sdk/MaybeSDK.scala:84]: Yeah, looks like those got mangled when they were copied from the similar Result/Either helpers. Have a PR in to fix this, currently building.

**Pull request merged**